_posts/2018-01-04-notification-channels-android-o.md

```diff
@@ -0,0 +1,129 @@
+        notificationManager.notify(PM_NOTIFICATION_ID, notification)
+```
+The part of the code above that sets the channel is `.setChannelId()`. Without this line, if you're targeting Android O and above, your notification will not be shown to the user.
+
+### Updating/reading notification channels settings
+
+After creating notification channels, the ability to control is left to the user. They can turn off notifications, change priority, lights & vibration settings etc for any channel they desire. However, it is possible to programmatically read the current channel settings. You may need this, so that you can present the status of the notification channel to your user - for example, if the user has turned off notification for a certain group
+chat, you may display a "mute" icon next to the group details or somewhere it's visible to the user. 
+
+It's important not to make this intrusive in anyway or obstructing them from going about their normal activities in the app because it defeats the purpose of giving users the power to control the settings for the channel.
+
+To read the notification channel details, you need to get a reference to the notification channel, and query the settings you desire.
+
+```kotlin
+val pmNotificationChannel = notificationManager.getNotificationChannel(GROUP_CHAT_CHANNEL_ID)
+
+val channelIsBlocked = pmNotificationChannel.importance == NotificationManager.IMPORTANCE_NONE
+
+if (channelIsBlocked) {
+	// do something unintrusive to make user aware that they blocked the notification
+	// you should also provide a way to make them go to the channel settings
+}
+```
+
+### Deleting notification channels
+So, assuming that a user has unsubscribed from a previous conversation that a channel was created for, and you have to delete the channel. Deleting is quite straightforward, to do this, one needs to do something like:
+
+```kotlin
+notificationManager.deleteNotificationChannel(PRIVATE_MESSAGES_CHANNEL_ID)
+```
+
+### Further reading
+Here are some materials that may come in handy for further reading of notification channels:
+* [https://developer.android.com/guide/topics/ui/notifiers/notifications.html#ManageChannels](https://developer.android.com/guide/topics/ui/notifiers/notifications.html#ManageChannels)
+* [https://codelabs.developers.google.com/codelabs/notification-channels-kotlin/#0](https://codelabs.developers.google.com/codelabs/notification-channels-kotlin/#0)
+
+Summary
+-----
+
+In conclusion, the new notification channels in Android provide an improved experience for users - in terms of gaining more control over notifications, and this also helps us developers to be more decent in their use of notifications, and at the end of the day, it's a win-win situation.
+
+Finally, remember that come August, 2018, new apps and app updates will require you to target Android O (API level 26) or higher, and as a result will require you to implement notification channels if you show notifications in your app.
```

Comment:
"come August, 2018"
Not sure about this... New apps, August. Existing apps/app updates, November

http://www.androidpolice.com/2017/12/19/play-store-require-new-updated-apps-target-recent-api-levels-distribute-native-apps-64-bit-support/